src/orders/src/main/java/mushop/orders/services/MessagingService.java

```diff
@@ -0,0 +1,96 @@
+package mushop.orders.services;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.nats.client.Connection;
+import io.nats.client.Dispatcher;
+import io.nats.client.Message;
+import io.nats.client.Nats;
+import mushop.orders.entities.CustomerOrder;
+import mushop.orders.repositories.CustomerOrderRepository;
+import mushop.orders.values.OrderUpdate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+@Service
+public class MessagingService {
+    @Autowired
+    private CustomerOrderRepository customerOrderRepository;
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private Connection nc;
+    private final String MUSHOP_ORDERS_SUBJECT;
+    private final String MUSHOP_SHIPMENTS_SUBJECT;
+    private final String NATS_URL;
+
+    public MessagingService(@Value("nats://${mushop.messaging.host}:${mushop.messaging.port}") String NATS_URL,
+                            @Value("${mushop.messaging.subjects.orders}") String MUSHOP_ORDERS_SUBJECT,
```

Comment:
is it a convention in Java to use uppercase names in ctor/methods? (I'd use `natsUrl`, `mushopOrdersSubject`,.. instead)

Reply:
Yep - will fix.